Models/DraggableContainer.py

```diff
@@ -140,10 +138,15 @@ def paintEvent(self, e: QPaintEvent):
     def mousePressEvent(self, e: QMouseEvent):
         self.position = QPoint(e.globalX() - self.geometry().x(), e.globalY() - self.geometry().y())
 
-        # Undo related
-        self.old_x = e.globalX()
-        self.old_y = e.globalY()
-        self.old_state = {'type':'object','action':'move','name':self.name,'x':self.old_x,'y':self.old_y}
+        #here
+        if self.mode == Mode.MOVE:
+            # Undo related
+            self.old_x = e.globalX()
+            self.old_y = e.globalY()
+            self.old_state = {'type':'object','action':'move','name':self.name,'geom':self.geometry()}
+        elif self.mode == Mode.RESIZEB or self.mode == Mode.RESIZEBL or self.mode == Mode.RESIZEBR or self.mode == Mode.RESIZEL or self.mode == Mode.RESIZER or self.mode == Mode.RESIZET or self.mode == Mode.RESIZETL or self.mode == Mode.RESIZETR:
```

Comment:
this can prob be 'elif self.mode != Mode.NONE' right?

Reply:
yeah it could work like that but if we add other modes in future, then it might be true when it does not have to

Reply:
hmm good point